File: src/main/java/net/casualuhc/uhcmod/command/FullBrightCommand.java
package net.casualuhc.uhcmod.command;

import com.mojang.brigadier.CommandDispatcher;
import net.casualuhc.uhcmod.managers.PlayerManager;
import net.casualuhc.uhcmod.utils.data.PlayerExtension;
import net.casualuhc.uhcmod.utils.data.PlayerFlag;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;

import static net.minecraft.server.command.CommandManager.literal;

public class FullBrightCommand {
	public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
		dispatcher.register(literal("fullbright").executes(c -> {
			ServerPlayerEntity player = c.getSource().getPlayerOrThrow();
			PlayerExtension extension = PlayerExtension.get(player);
			extension.toggleFlag(PlayerFlag.FULL_BRIGHT_ENABLED);
			PlayerManager.updateFullBright(player, false);
			return 1;
		}));
	}
}

File: src/main/kotlin/net/casualuhc/uhcmod/commands/FullbrightCommand.kt
package net.casualuhc.uhcmod.commands

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import net.casualuhc.uhcmod.extensions.PlayerFlag.FullBright
import net.casualuhc.uhcmod.extensions.PlayerFlagsExtension.Companion.flags
import net.casualuhc.uhcmod.util.Texts
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands

object FullbrightCommand: Command {
    override fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
            Commands.literal("fullbright").executes {
                this.toggle(it)
            }.then(
                Commands.literal("enabled").executes {
                    this.toggle(it, true)
                }
            ).then(
                Commands.literal("disabled").executes {
                    this.toggle(it, false)
                }
            )
        )
    }

    private fun toggle(context: CommandContext<CommandSourceStack>): Int {
        val player = context.source.playerOrException
        return this.toggle(context, !player.flags.has(FullBright))
    }

    private fun toggle(context: CommandContext<CommandSourceStack>, set: Boolean): Int {
        context.source.playerOrException.flags.set(FullBright, set)
        context.source.sendSystemMessage(Texts.UHC_FULL_BRIGHT.generate(set))
        return 1
    }
}